src/arch/xtensa/CMakeLists.txt

```diff
@@ -107,7 +107,7 @@ function(sof_add_ld_script binary_name script_name)
 	endforeach()
 
 	add_custom_command(OUTPUT ${lds_out}
-		COMMAND cat ${lds_in} | ${CMAKE_C_COMPILER} -E -P ${iflags} - > ${lds_out}
+		COMMAND ${CMAKE_C_COMPILER} -E -P ${iflags} ${lds_in} > ${lds_out}
```

Comment:
I guess now it doesn't work on linux. Shouldn't there be something like
```
${CMAKE_C_COMPILER} -E -P ${iflags} - ${lds_in} > ${lds_out}
```
or
```
${CMAKE_C_COMPILER} -E -P ${iflags} - < ${lds_in} > ${lds_out}
```
?

Reply:
It is not the Linux, it is rather the gcc as I understand it (xcc looks good on Linux). Although it works in -E mode. Let's try with <.